src/ORM/DataObject.php

```diff
@@ -1208,7 +1210,7 @@ public function merge($rightObj, $priority = 'right', $includeRelations = true,
                     $leftComponents = $leftObj->getManyManyComponents($relationship);
                     $rightComponents = $rightObj->getManyManyComponents($relationship);
                     if ($rightComponents && $rightComponents->exists()) {
-                        $leftComponents->addMany($rightComponents->column('ID'));
+                        $leftComponents->addMany($rightComponents->sort(null)->column('ID'));
```

Comment:
```suggestion
                        $leftComponents->addMany($rightComponents->column('ID'));
```
This one I'm not sure about because it's a bit of a weird method, though it feels like there's a decent chance of regressions

There doesn't seem like we actually use this method much, though have a look at SiteTree::BackLinkTracking() - it seems like we'd want to retain sort order here e.g. sort by Title

Reply:
> it feels like there's a decent chance of regressions

Can you please explain what your thinking is there?

It looks safe to me. `addMany()` loops through the items and passes them to `add()`. `ManyManyList::add()` does either an SQL `UPDATE` or `INSERT` depending on whether there's existing records for that item... but it doesn't affect the sort order of the actual list, because iterating the list still requires executing a new `SELECT` query which has its own sort applied to it.

I don't see how `SiteTree::BackLinkTracking()` is relevant.

Reply:
Opps - noticed that SiteTree::BackLinkTracking() uses ArrayList::merge(), I mistakenly thought it related to this method

Yes agree this will be fine unsorted